tests/integration/llm/sagemaker-endpoint-tests.py

```diff
@@ -98,75 +104,166 @@
     }
 }
 
-ROLE = "arn:aws:iam::185921645874:role/AmazonSageMaker-ExeuctionRole-IntegrationTests"
-DEFAULT_INSTANCE_TYPE = "ml.g5.12xlarge"
-DEFAULT_PAYLOAD = {"inputs": "Deep Learning is"}
+MME_CONFIGS = {
+    "deepspeed-mme": {
+        "models": [{
+            "model_id": "EleutherAI/gpt-neo-2.7B",
+            "model_kwargs": {
+                "dtype": "fp16",
+                "number_of_partitions": 1,
+            }
+        }, {
+            "model_id": "s3://djl-llm/opt-1.3b/",
+            "model_kwargs": {
+                "dtype": "fp16",
+                "number_of_partitions": 1,
+            }
+        }],
+        "instance_type":
+        "ml.g5.8xlarge",
+        "image_uri":
+        "125045733377.dkr.ecr.us-east-1.amazonaws.com/djl-serving:0.22.1-deepspeed",
+    }
+}
+
+
+def get_sagemaker_session(default_bucket=DEFAULT_BUCKET,
+                          default_bucket_prefix=None):
+    return sagemaker.session.Session(
+        boto3.session.Session(),
+        default_bucket=default_bucket,
+        default_bucket_prefix=default_bucket_prefix)
+
+
+def delete_s3_test_artifacts(sagemaker_session):
+    bucket = sagemaker_session.get_default_bucket()
+    prefix = sagemaker_session.get_default_bucket_prefix()
+    s3 = boto3.resource("s3")
+    s3.Bucket(bucket).objects.filter(Prefix=prefix).delete()
+
+
+def get_name_for_resource(name):
+    cleaned_name = ''.join(filter(str.isalnum, name))
+    base_name = "sm-integration-test-{}".format(cleaned_name)
+    return unique_name_from_base(base_name)
+
+
+def mme_test(name):
+    config = MME_CONFIGS.get(name)
+    session = get_sagemaker_session(default_bucket_prefix="mme-tests")
+    models = config.get("models")
+    created_models = []
+    mme = None
+    predictor = None
+    try:
+        for model_config in models:
+            model = DJLModel(model_config.get("model_id"),
+                             ROLE,
+                             image_uri=config.get("image_uri"),
+                             name=get_name_for_resource(
+                                 model_config.get("model_id") + '-mme'),
+                             sagemaker_session=session,
+                             **model_config.get("model_kwargs"))
+            model.create()
+            created_models.append(model)
+
+        mme = MultiDataModel(get_name_for_resource(name),
```

Comment:
How does this map to the created models?

Reply:
MME works by having models registered under a specific s3 bucket and prefix. All tar.gz files under there are models that can be loaded by the endpoint. 

Above I create a few models in the same s3 path that the MME is created. Then invoking the mme you pass in the artifact name relative to the s3 bucket/prefix. 

It's a bit strange, but for now this is how it works with the PySDK.

Reply:
Oh interesting. Good to know